tests/test_grid.py

```diff
@@ -0,0 +1,294 @@
+"""Test chi^2 gridding routines"""
+import concurrent.futures
+import io
+import multiprocessing
+import os
+import pytest
+
+import astropy.units as u
+import numpy as np
+
+import pint.config
+import pint.gridutils
+import pint.models.parameter as param
+from pint.fitter import WLSFitter
+from pint.models.model_builder import get_model_and_toas
+
+# for multi-core tests, don't use all available CPUs
+ncpu = 2
+
+
+def test_grid_singleprocessor():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = WLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 3 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 3 * f.model.F0.uncertainty,
+        15,
+    )
+    F1 = np.linspace(
+        f.model.F1.quantity - 3 * f.model.F1.uncertainty,
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        17,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1), ncpu=1)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_multiprocessor():
```

Comment:
Why not test the multiprocessor values to see if they're the same as the single-processor values?

Reply:
I did see some very small differences (at the 1e-6 level).  I didn't track down the origin.